### DISCUSSION TITLE
Open Policy Agent vs Casbin - Qinng

### DISCUSSION THREAD

**Discussion by giscus** [qingwave/qingwave.github.io#72] [Announcements]
# Open Policy Agent vs Casbin - Qinng

大型项目中基本都包含有复杂的访问控制策略，特别是在一些多租户场景中，例如Kubernetes中就支持RBAC，ABAC等多种授权类型。在Golang中目前比较热门的访问控制框架有Open Policy Agent与

https://qingwave.github.io/openpolicyagent-vs-casbin/

**Comment by acechef** [comment]
拒绝非用户用户访问api资源

**Reply by acechef** [reply]
这里读不通

**Reply by qingwave** [reply]
多谢提醒，这里应该是”拒绝未认证用户访问”
